### ISSUE TITLE
What are the User Stories / Use Cases wrt to the UI?

### ISSUE THREAD

**Issue by erantapaa** [erantapaa/tennis-match-scorer#9]
A lot of this assignment seems to concern itself with the UI.

What are the [user stories / use cases][1] associated with the UI? What is your impression of how the UI is suppose to work?

The assignment talks about "sav[ing] intermediate awards of ranking points and prize money" and "updat[ing] each player’s totals". Does that mean data about the players are stored in a file and updated?

it also mentions a "menu system". What options do you envision are going to be on that menu?

[1]: https://www.stellman-greene.com/2009/05/03/requirements-101-user-stories-vs-use-cases/

**Comment by toby2-wyllie** [comment]
For the UI I believe that we should be able to select a tournament > select a round and view results.

There should also be an option to view an overall points ranking table and displaying the total prize money earned

**Comment by toby2-wyllie** [comment]
Start menu
![img_7298](https://user-images.githubusercontent.com/34072700/33350575-66601d3a-d497-11e7-8e6a-0d33b685c3b6.JPG)

Once tournament has been chosen
![img_7299](https://user-images.githubusercontent.com/34072700/33350531-338dac9c-d497-11e7-87db-7224d64f02e6.JPG)

Overall Ranking for tournament 
![img_7300](https://user-images.githubusercontent.com/34072700/33350540-3f7421c6-d497-11e7-9b4b-bcf47ae13ddc.JPG)

**Comment by toby2-wyllie** [comment]
I understand that doing this with a GUI would be more complex but that's the best way I can describe what I want to do. (Man I wish I was actually good at programming)

**Comment by erantapaa** [comment]
Does it have to be  GUI? How about a terminal-based menu system?

The UI aspect of this project is 10x more code than the logic for determining ranking and prize money.

**Comment by toby2-wyllie** [comment]
Yeah a terminal-based menu system would be great! As long as it has all the same functions as what  I've sketched out roughly.

**Comment by erantapaa** [comment]
Ok - we have to determine what the _commands_ are for the system.

The system keeps track of match results and the standings of all players in the 8 tournaments (4 men's and 4 women's.) The main function is to enter match results. The system updates standings and if necessary rankings and prize money. The system also allows you to view current standings.

The commands to the system are:

* Record a match result. That is, for a specific _tournament_ in a specific _round_ between two _players_ record a match _score_. Update a player's point total, ranking, prize money as needed.

* List information about a tournament. Some useful queries that come to mind are: what round is it in, what players are still in the round, etc.

* List information about a player, e.g. What tournaments the player is in, current ranking, prize winnings, etc.

* Quit.

Reading and applying a bunch of match results from a file is just repeating the "record match result" command multiple times.

After every change to the system's data you need to save it so that you don't lose any results.

So the basic program loop will look like:

    sysdata = read system's data from the save file
    while True:
        cmd = getCommand()
        if cmd is "quit": break
        if cmd is record result:
            update sysdata with the new result
            save sysdata to the save file
        else if cmd is list player info:
            print a report about a player
        else if cmd is list tournament info:
            print a report about a specific tournament
        else:
            print "Huh?"

Does that look about right?

**Comment by erantapaa** [comment]
I added `basicloop.py` to the repo which demonstrates a basic command loop.

You should be able to do this at the shell:

    shell-prompt$ python3 ./basicloop.py
    $ list-matches
    Tournament Round PlayerA    PlayerB    ScoreA ScoreB
    $ result FOO 1 MP1 3 Mp2 4
    $ list-matches
    Tournament Round PlayerA    PlayerB    ScoreA ScoreB
    FOO        1     MP1        Mp2        3 4
    $ quit

    shell-prompt$ python3 ./basicloop.py
    $ list-matches
    Tournament Round PlayerA    PlayerB    ScoreA ScoreB
    FOO        1     MP1        Mp2        3 4

So the entered match was saved between runs. It doesn't do much else, but you should see how to add commands. You'll want to add data validation.

The data is stored in the file `sysdata.dat`. To start over just remove that file.
